tmt/base.py

```diff
@@ -2187,6 +2214,9 @@ def go(self) -> None:
                         color='yellow', shift=1)
                     abort = True
                     return
+                # Source the plan environment file after prepare step
+                if isinstance(step, tmt.steps.prepare.Prepare):
+                    self._source_plan_env_file()
```

Comment:
I suppose `execute` step may add variables to the file as well, hoping to expose them to `finish` phases. Re-including the file once more after `execute` would allow that.

Reply:
I changed it so the file is sourced after the execute step as well. However, I encountered a problem with the `environment_file_to_dict` function I used because it was cached. So, I used the `__wrapped__` attribute to ignore the cache. Is there a better way to do that, or is this approach okay?

Reply:
`__wrapped__` seems to be fine. There's `clear_cache()` but that seems to be a hammer way too big.